codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java

```diff
@@ -264,27 +269,35 @@ public ChannelFuture writeSettingsAck(ChannelHandlerContext ctx, ChannelPromise
     }
 
     @Override
-    public ChannelFuture writePing(ChannelHandlerContext ctx, boolean ack, ByteBuf data,
-            ChannelPromise promise) {
-        boolean releaseData = true;
-        SimpleChannelPromiseAggregator promiseAggregator =
+    public ChannelFuture writePing(ChannelHandlerContext ctx, boolean ack, ByteBuf data, ChannelPromise promise) {
+        final SimpleChannelPromiseAggregator promiseAggregator =
                 new SimpleChannelPromiseAggregator(promise, ctx.channel(), ctx.executor());
         try {
             verifyPingPayload(data);
             Http2Flags flags = ack ? new Http2Flags().ack(true) : new Http2Flags();
+            int payloadLength = data.readableBytes();
             ByteBuf buf = ctx.alloc().buffer(FRAME_HEADER_LENGTH);
```

Comment:
buffer should be released in the catch below right?

Reply:
no because it is written to `ctx` which transfers reference count ownership.